rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -2344,6 +2345,71 @@ def _check_gc_data_files(self):
                 _logger.info(f"Will try to reconcile sample ids: {needs_data_file_reconciliation}")
 
 
+class UpdateMissingFiles(ToolBase):
+    def __init__(self, args, gcp_env: GCPEnvConfigObject):
+        super().__init__(args, gcp_env)
+        self.member_dao = None
+        self.data_file_dao = None
+        self.metrics_dao = None
+
+    def run(self):
+        self.gcp_env.activate_sql_proxy()
+
+        self.member_dao = GenomicSetMemberDao()
+        self.metrics_dao = GenomicGCValidationMetricsDao()
+        self.data_file_dao = GenomicGcDataFileDao()
+
+        samples_list = []
+        updated_count = 0
+
+        if self.args.file_path:
+            if os.path.exists(self.args.file_path):
+                with open(self.args.file_path, encoding='utf-8-sig') as file:
+                    lines = file.readlines()
+                    for line in lines:
+                        samples_list.append(line.strip())
+            else:
+                _logger.error(f"Unable to open {self.args.file_path}")
+                return 1
+
+        members_to_update = self.member_dao.get_array_members_files_available(samples_list)
+        members_to_update.extend(self.member_dao.get_wgs_members_files_available(samples_list))
+        if not self.args.dryrun:
+            for member in members_to_update:
+                updated = self._update_member(member)
+                if updated:
+                    updated_count += 1
+        else:
+            _logger.info(f"Will update {len(members_to_update)} samples")
+        _logger.info(f"Found {len(members_to_update)} members to update. Updated {updated_count}.")
+
+    def _update_member(self, member: GenomicSetMember) -> bool:
```

Comment:
This function updates the metric records, might be a good idea to rename is `_update_metrics`